packages/nextcloud_test/lib/src/test_target/test_target.dart

```diff
@@ -6,28 +6,40 @@ import 'package:cookie_store/cookie_store.dart';
 import 'package:meta/meta.dart';
 import 'package:neon_http_client/neon_http_client.dart';
 import 'package:nextcloud/nextcloud.dart';
-import 'package:nextcloud_test/nextcloud_test.dart';
 import 'package:nextcloud_test/src/fixtures.dart';
+import 'package:nextcloud_test/src/models/models.dart';
 import 'package:nextcloud_test/src/proxy_http_client.dart';
 import 'package:nextcloud_test/src/test_target/docker_container.dart';
 import 'package:nextcloud_test/src/test_target/local.dart';
 import 'package:version/version.dart';
 
 /// Factory for creating [TestTargetInstance]s.
+
 abstract class TestTargetFactory<T extends TestTargetInstance> {
+  /// The instance of the [TestTargetFactory].
+  static final TestTargetFactory instance = TestTargetFactory._create();
+
   /// Creates a new [TestTargetFactory].
-  static TestTargetFactory create() {
-    final url = Platform.environment['URL'];
+  static TestTargetFactory _create() {
     final dir = Platform.environment['DIR'];
-    if (url != null && url.isNotEmpty && dir != null && dir.isNotEmpty) {
-      return LocalFactory();
+    final url = Platform.environment['URL'];
+    Uri? uri;
+    if (url != null) {
+      uri = Uri.tryParse(url);
+    }
+
+    if (uri != null && dir != null && dir.isNotEmpty) {
```

Comment:
The hard fail was on purpose. If you give it faulty config now it will not complain and just use docker.

Reply:
So you mean I should throw if the url can not be parsed?

Reply:
Yes, same for empty DIR variable (that wasn't checked even before I think)